### ISSUE TITLE
`needless_pass_by_value` flags underscore-prefixed arguments

### ISSUE THREAD

**Issue by rileyshahar** [rust-lang/rust-clippy#7295]
I'm not actually certain if this is the best behavior of this lint, but I figured I'd open an issue.

The following code triggers clippy's `needless_pass_by_value` lint:
```rust
fn foo(_bar: Vec<i32>) {}
```

However, the underscore in `_bar` has already told the compiler that I intend to use `bar` later in the unimplemented parts of the function; hence why the compiler doesn't give this code an unused variable warning. I think perhaps the underscore should also turn off the needless pass by value lint: if this variable is being used while planning the program's structure and the compiler has already been told that its usage is incomplete, then clippy can't know whether or not the final usage will require the argument by value.

In fact, this warning could cause some significant issues in this instance, if it causes someone to change the signature to accept the argument by reference, but they later turn out to need it by value, and so they've designed their program on the stubbed-out version passing the borrow checker because of the incorrect signature. I think this is unlikely & this change is more fixing an annoyance than anything, but it seems to me like it would eliminate far, far more false positives than it induces false negatives.
